src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -223,6 +223,11 @@ private void UpdateEndpoints()
         {
             var newPathSegments = routePattern.PathSegments.ToList();
             var hasLinkGenerationEndpoint = false;
+            var nonParameterPolicyValues = routePattern.ParameterPolicies
+                .Where(p => routePattern.GetParameter(p.Key ?? string.Empty) == null && p.Value.Count > 0 && p.Value.First().ParameterPolicy != null) // Only GetParameter is required. Extra is for safety
+                .Select(p => new KeyValuePair<string, object>(p.Key, p.Value.First().ParameterPolicy)) // Can only pass a single non-parameter to RouteParameter
+                .ToArray();
+            var nonParameterPolicies = RouteValueDictionary.FromArray(nonParameterPolicyValues);
```

Comment:
How do the parameter-related policies get injected? 

For instance, do we have tests for `"{controller}/{action}/{id:int?}"` as well as `"{controller}/{action}/{id?}" + new { id = new  IntRouteConstraint(), }`?

Reply:
NVM I see you have a test. 

What about a case where `id` has an inline **and** out of line constraint?

Reply:
Done

Reply:
They should be effected because they're both associated with a parameter, but I've updated the test to include a parameter with both.